lib/ot.js

```diff
@@ -4,46 +4,49 @@
 // delete operations.
 
 var types = require('./types').map;
+var ShareDBError = require('./error');
+
+var ERROR_CODE = ShareDBError.CODES;
 
 // Returns an error string on failure. Rockin' it C style.
 exports.checkOp = function(op) {
   if (op == null || typeof op !== 'object') {
-    return {code: 4004, message: 'Missing op'};
+    return {code: ERROR_CODE.ERR_OT_OP_BADLY_FORMED, message: 'Op must be an object'};
   }
 
   if (op.create != null) {
     if (typeof op.create !== 'object') {
-      return {code: 4006, message: 'create data must be an object'};
+      return {code: ERROR_CODE.ERR_OT_OP_BADLY_FORMED, message: 'Create data must be an object'};
     }
     var typeName = op.create.type;
     if (typeof typeName !== 'string') {
-      return {code: 4007, message: 'Missing create type'};
+      return {code: ERROR_CODE.ERR_OT_OP_BADLY_FORMED, message: 'Missing create type'};
     }
     var type = types[typeName];
     if (type == null || typeof type !== 'object') {
-      return {code: 4008, message: 'Unknown type'};
+      return {code: ERROR_CODE.ERR_DOC_TYPE_NOT_RECOGNISED, message: 'Unknown type'};
```

Comment:
Let's change this to `OP_BADLY_FORMED` to fit in with everything else in this  function.

Reply:
Actually let's leave this as-is.